transport/control.go

```diff
@@ -236,9 +237,10 @@ func (f *inFlow) loanWindowSpace(n uint32) uint32 {
 		grpclog.Fatalf("pre-consuming window space while there is pre-consumed window space still outstanding")
 	}
 	f.loanedWindowSpace = n
+	f.pendingUpdate += f.loanedWindowSpace
```

Comment:
We shouldn't add loanedWindowSpace to pendingUpdate since that would in some cases cause pendingUpdate + pendingData > limit since we have loaned over the window limit.